src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs

```diff
@@ -35,7 +124,7 @@ internal static class DiagnosticHelper
         /// </param>
         /// <param name="messageArgs">Arguments to the message of the diagnostic.</param>
         /// <returns>The <see cref="Diagnostic"/> instance.</returns>
-        public static Diagnostic Create(
+        public static DiagnosticWrapper Create(
```

Comment:
@mavasani All analyzers will have to do the null check. I wanted to minimize changes to analyzers.

And an extension method named `ReportDiagnostic` will not picked up unless the parameter type is different (the compiler will prefer the existing ReportDiagnostic over the extension)
And Adding an extension with a new name will lead to updates in all analyzers.

Reply:
Thanks, so we are basically trading off the cleanest API to avoid making tons of changes in the code base. That seems reasonable to me, but I'll let others in @dotnet/roslyn-analysis share their views.

Reply:
Meant to tag @dotnet/roslyn-ide